plugins/csv_dataset.cc

```diff
@@ -575,7 +575,13 @@ struct CsvDataset::Itl: public TabularDataStore {
         filename = config.dataFileUrl.toString();
         
         // Ask for a memory mappable stream if possible
-        ML::filter_istream stream(filename, { { "mapped", "true" } });
+        ML::filter_istream stream;
+        try {
+            stream.open(filename, { { "mapped", "true" } });
+        }
+        catch (const std::exception & e) {
+             throw HttpReturnException(400, "dataFileUrl pointed to an empty file");
```

Comment:
Does it make more sense to create empty datasets for empty files? I guess I
can see that... Non-existent files should cause an exception though, right?

Is it possible to specify a csv dataset with no columns? Should we return
something different if headers are specified and the file is empty vs no
headers and empty file!

On Monday, 25 January 2016, Jeremy Barnes <notifications@github.com> wrote:

> In plugins/csv_dataset.cc
> <https://github.com/mldbai/mldb/pull/100#discussion_r50762565>:
>
> > @@ -575,7 +575,13 @@ struct CsvDataset::Itl: public TabularDataStore {
> >          filename = config.dataFileUrl.toString();
> >
> >          // Ask for a memory mappable stream if possible
> > -        ML::filter_istream stream(filename, { { "mapped", "true" } });
> > +        ML::filter_istream stream;
> > +        try {
> > +            stream.open(filename, { { "mapped", "true" } });
> > +        }
> > +        catch (const std::exception & e) {
> > +             throw HttpReturnException(400, "dataFileUrl pointed to an empty file");
>
> Finally, the filter_istream should not fail to map an empty file; it
> should return a zero length mapping. Once that is fixed, the check for zero
> length can be made elsewhere (and @nicolaskruchten
> <https://github.com/nicolaskruchten> does it make sense to disallow an
> empty csv file? One with no rows and no columns is empty, but a valid (if
> useless) CSV file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mldbai/mldb/pull/100/files#r50762565>.
>


-- 
Nicolas Kruchten, Eng  |  Staff Software Engineer at Datacratic  |
http://datacratic.com/